### DISCUSSION TITLE
Sending "power 1" two times results in toggle instead of ignoring second command

### DISCUSSION THREAD

**Discussion by some-rando** [arendst/Tasmota#12437] [Q&A] [answered]
Greetings. I have a US-SD-KN01 from Martin Jerry with Tasmota 9.5.0 installed. When commanding it to turn on or off TWO TIMES with either `power 0` in Tasmota console, or via Node-RED, or via Home Assistant service, the light will **toggle** instead of ignoring the second on (or 2x off) command like I would expect. In other words, if the light is off and I send `light.turn_off` the light turns ON!

Since the terms "power" or "toggle" are super generic, my google-fu isn't helping like usual. Does anyone have an idea of where to start searching for a solution to this?

For a more concrete example: I have a Node-RED flow that checks a boolean_input in Home Assistant for "Heat Wave Mode" that I've created to keep the lights on for shorter times when it's super hot out. Normally I can send a command to a light an unlimited number of times and if the command doesn't introduce a change then the light doesn't change. So sending `power_on` once will turn on the light. Sent again and it will turn off the light. Sent a third time it will turn on the light. And so on, toggling each time.

I've also tried manually typing `power 1` into the Tasmota web console and it does the same toggle. And sending `light.turn_on` via Home Assistant services also toggles. For completeness, sending `light.toggle` toggles the light on and off as expected.

`status` gives me `MQT: stat/Kitchen_Recessed_Light/STATUS = {"Status":{"Module":54,"DeviceName":"Kitchen_Recessed_Light","FriendlyName":["Kitchen_Recessed_Light"],"Topic":"Kitchen_Recessed_Light","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}`

Any thoughts? Thanks for any help.

**Comment by sfromis** [comment, marked as answer]
Unfortunately, Tasmota has no direct control for TuyaMCU devices, with the other MCU doing whatever it fancies. Studying the log when it happens might provide clues about why. In general, if you do not want to handle added state management backend, you can also do it via rules, and then send `Event` commands to trigger rules for the action to be performed, including logic to avoid double takes.

**Reply by some-rando** [reply]
I hadn't realized it was a TuyaMCU device, or forgot! Thanks for the info and good ideas to point me in the right direction for a solution. I appreciate it.
